src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -90,14 +91,26 @@ export default class EthereumBlockchainService implements BlockchainService {
   }
 
   /**
-   * Takes current gas price and attempt number and returns new gas price with 10% increase per attempt
-   * @param currentGas
+   * Takes current gas price and attempt number and returns new gas price with 10% increase per attempt.
+   * If this isn't the first attempt, also ensures that the new gas is at least 10% greater than the
+   * previous attempt's gas, even if the gas price on chain has gone down since then. This is because
+   * retries of the same transaction (using the same nonce) need to have a gas price at least 10% higher
+   * than any previous attempts or the transaction will fail.
+   * @param currentGasEstimate
    * @param attempt
+   * @param previousGas
    */
-  static increaseGasPricePerAttempt(currentGas: BigNumber, attempt: number): BigNumber {
-    const tenPercent = currentGas.div(10)
+  static increaseGasPricePerAttempt(currentGasEstimate: BigNumber, attempt: number, previousGas: BigNumberish | undefined): BigNumber {
```

Comment:
Why "BigNumberish" here? instead of BigNumber

Reply:
Because that's what `TransactionRequest.gasPrice` is :frowning_face: https://github.com/ethers-io/ethers.js/blob/6c43e20e7a68f3f5a141c74527ec63d9fe8458be/packages/abstract-provider/src.ts/index.ts#L24
I tried converting it at the call site, but then you have to put the handling for `undefined` there too which complicated things.  This seemed the most straightforward.